### PR TITLE
Dedicated threadpool supervisor

### PR DESCRIPTION
**Pull request by rogeralsing**
This adds the new DedicatedThreadPool supervisor.

Conceptual implementation:

The supervisor uses a `System.Threading.Timer` to poll workers for a status.
The status is updated by scheduling an action to each worker, and when that action executes, the worker status is updated.
if the worker status have not been updated untill next tick, the thread is considered dead.
If the thread is considered dead, the supervisor will force termination of the underlying thread and request a new one.

TODO: Add tests for the supervisor
TODO: fix the now broken End of the world test

A severe bug in pool ctor is corrected too

### CODE REVIEW

File: src/core/Helios.DedicatedThreadPool/Helios.Concurrency.DedicatedThreadPool.cs [hunk 1]
```diff
@@ -194,20 +194,54 @@ private void RequestWorker()
...
+                    }
+                    //TODO: make the poll time out configurable
+                }, null, TimeSpan.FromSeconds(5), TimeSpan.FromSeconds(5));
+            }
+        }
        
         public DedicatedThreadPool(DedicatedThreadPoolSettings settings)
         {
             Settings = settings;
-            Workers = Enumerable.Repeat(new WorkerQueue(), settings.NumThreads).ToArray();
```
Comment (rogeralsing) [inline src/core/Helios.DedicatedThreadPool/Helios.Concurrency.DedicatedThreadPool.cs:206]: This code will repeat the _same_ worker queue NumThread times.
The intention is to create a new worker queue for each repetition. this have been fixed

**Comment by rogeralsing** [comment]
There are more work to be done on this one, just wanted to check with you if you think this is the correct approach to solve the problem.

It does fix the problem of zombie processes while being essentially free in terms of performance.

**Comment by Aaronontheweb** [comment]
Going forward could you repoint your Fork to https://github.com/helios-io/DedicatedThreadPool? That way our CI server can pick it up.

File: src/core/Helios.DedicatedThreadPool/Helios.Concurrency.DedicatedThreadPool.cs (hunk 1 above)
Comment (Aaronontheweb) [inline src/core/Helios.DedicatedThreadPool/Helios.Concurrency.DedicatedThreadPool.cs:206]: Ah, nice catch - thanks!

**Comment by Aaronontheweb** [comment]
Pulled this into my `dev` branch.

**Pull request closed without merging**